src/main/java/com/arpnetworking/kairosdb/HistogramDataPointImpl.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2017 SmartSheet.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.kairosdb;
+
+import org.json.JSONException;
+import org.json.JSONWriter;
+import org.kairosdb.core.datapoints.DataPointHelper;
+
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * DataPoint that represents a Histogram.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class HistogramDataPointImpl extends DataPointHelper implements HistogramDataPoint {
+    private static final String API_TYPE = "histogram";
+    private final int _precision;
+    private final TreeMap<Double, Integer> _map;
+    private final double _min;
+    private final double _max;
+    private final double _mean;
+    private final double _sum;
+
+    /**
+     * Public constructor.
+     *
+     * @param timestamp the timestamp.
+     * @param precision bucket precision, in bits
+     * @param map the bins with values
+     * @param min the minimum value in the histogram
+     * @param max the maximum value in the histogram
+     * @param mean the mean value in the histogram
+     * @param sum the sum of all the values in the histogram
+     */
+    public HistogramDataPointImpl(final long timestamp, final int precision, final TreeMap<Double, Integer> map, final double min,
```

Comment:
Please chop these; one per newline with the first on a new line double indented.